gtsam/linear/JacobianFactor.cpp

```diff
@@ -515,7 +515,7 @@ Vector JacobianFactor::error_vector(const VectorValues& c) const {
 double JacobianFactor::error(const VectorValues& c) const {
   Vector e = unweighted_error(c);
   // Use the noise model distance function to get the correct error if available.
-  if (model_) return 0.5 * model_->distance(e);
+  if (model_) return model_->error(e);
```

Comment:
I rethought about it, and I think we still need the error function. Because the way error is calculated varies in different noise models (Gaussian, Robust, Constrained). If we do not have an "error" function inside the noise models, we'll need to implement it in the "error" function of "NoiseModelFactor".

In the document, we already used `d` to represent the new reweighted Mahalanobis distance

Reply:
2nd

Reply:
Yetong - we agreed the code would reflect the math. What symbol does “error” correspond to? If there is no corresponding symbol you have two choices: add a symbol in math, or remove this extra function :-)

Reply:
Got it, will add symbol to document

Reply:
Actually, I much prefer if you took a hard look at the document and how the robust (and quadratic) models are used in a factor (which *does* have an error) and my favorite outcome would be : "wow, we don't need this function at all!"

Reply:
I'm not sure about this. How about discuss about this in our meeting?